pkg/restore/restore.go

```diff
@@ -474,11 +485,41 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 			if !existingNamespaces.Has(mappedNsName) {
 				logger := ctx.log.WithField("namespace", nsName)
 				ns := getNamespace(logger, filepath.Join(dir, api.ResourcesDir, "namespaces", api.ClusterScopedDir, nsName+".json"), mappedNsName)
-				if _, err := kube.EnsureNamespaceExists(ns, ctx.namespaceClient); err != nil {
+				nsIsReady, err := kube.EnsureNamespaceExistsAndIsReady(ns, ctx.namespaceClient)
+				if err != nil {
 					addArkError(&errs, err)
 					continue
 				}
 
+				if !nsIsReady {
+					timeout := time.After(ctx.resourceTerminatingTimeout)
+					tick := time.Tick(1 * time.Second)
```

Comment:
Need a deferred stop of `tick` I believe

Reply:
Looks like `tick` is just the channel, not a `Ticker`. I'll redo it so we have a `Ticker` that we can actually stop.